src/test/java/bio/terra/service/snapshotbuilder/utils/SearchConceptsQueryBuilderTest.java

```diff
@@ -36,37 +36,37 @@ void buildSearchConceptsQuery(SqlRenderContext context) {
     SnapshotBuilderDomainOption domainOption =
         createDomainOption("Observation", 27, "observation", "observation_concept_id");
     String actual =
-        SearchConceptsQueryBuilder.buildSearchConceptsQuery(domainOption, "cancer")
+        new QueryBuilderFactory()
+            .searchConceptsQueryBuilder()
+            .buildSearchConceptsQuery(domainOption, "cancer")
             .renderSQL(context);
 
-    if (platformWrapper.isGcp()) {
-      assertThat(
-          "generated SQL for GCP is correct",
-          actual,
-          equalToCompressingWhiteSpace(
-              "SELECT c.concept_name, c.concept_id, COUNT(DISTINCT o.person_id) AS count "
-                  + "FROM concept AS c "
-                  + "JOIN observation AS o ON o.observation_concept_id = c.concept_id "
-                  + "WHERE (c.domain_id = 'Observation' "
-                  + "AND (CONTAINS_SUBSTR(c.concept_name, 'cancer') "
-                  + "OR CONTAINS_SUBSTR(c.concept_code, 'cancer'))) "
-                  + "GROUP BY c.concept_name, c.concept_id "
-                  + "ORDER BY count DESC "
-                  + "LIMIT 100"));
-    }
-    if (platformWrapper.isAzure()) {
-      assertThat(
-          "generated SQL for Azure is correct",
-          actual,
-          equalToCompressingWhiteSpace(
-              "SELECT TOP 100 c.concept_name, c.concept_id, COUNT(DISTINCT o.person_id) AS count "
-                  + "FROM concept AS c  JOIN observation AS o ON o.observation_concept_id = c.concept_id "
-                  + "WHERE (c.domain_id = 'Observation' "
-                  + "AND (CHARINDEX('cancer', c.concept_name) > 0 "
-                  + "OR CHARINDEX('cancer', c.concept_code) > 0)) "
-                  + "GROUP BY c.concept_name, c.concept_id "
-                  + "ORDER BY count DESC"));
-    }
+    assertThat(
+        "generated SQL for GCP is correct",
+        actual,
+        equalToCompressingWhiteSpace(
+            platformWrapper.choose(
+                () ->
+                    "SELECT c.concept_name, c.concept_id, COUNT(DISTINCT o.person_id) AS count "
```

Comment:
Could you instead write the tests so that the expected value is inside the `choose()` and the `assertThat()` is after? I think it would be clearer if there was only one `assertThat` instead of one for each branch.

Also, I think using a text block here instead of a string concatenation expression would be better. For an example, see `HierarchyQueryBuilderTest.generateQuery()`.